server/src/main/java/io/crate/planner/operators/EquiJoinDetector.java

```diff
@@ -65,6 +65,13 @@ public static boolean isEquiJoin(Symbol joinCondition) {
         return context.isHashJoinPossible;
     }
 
+    /**
+     * Returns true if the join condition is a simple equi join condition of the form, 'col1 = col2'.
+     */
+    public static boolean isPureEquiJoin(Symbol joinCondition) {
```

Comment:
`pure` may be a bit confusing, what about?

```suggestion
    public static boolean isTopLevelEquiJoin(Symbol joinCondition) {
```